pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -79,10 +107,17 @@ func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machi
 		machine:       machine,
 	})
 	if err != nil {
-		return fmt.Errorf("failed to create scope for machine %q: %v", machine.Name, err)
+		fmtErr := fmt.Errorf(scopeFailFmt, machine.Name, err)
+		return a.handleMachineError(machine, apierrors.InvalidMachineConfiguration("error decoding MachineProviderConfig: %v", fmtErr), updateEventAction)
```

Comment:
`s/error decoding MachineProviderConfig/error creating machine scope` would be more appropriate

Reply:
`InvalidMachineConfiguration` does not seem appropriate here